### ISSUE TITLE
Build issues with nix on mac

### ISSUE THREAD

**Issue by aviaviavi** [haskell-infra/www.haskell.org#122]
When running `buildAndWatch` on mac, I am getting some build errors from nix. `./buildAndWatch cabal` works fine. 

```
~/d/f/www.haskell.org ❯❯❯ ./buildAndWatch                                                                                  ✘ 130
error: cannot coerce null to a string, at /Users/avi/dev/forks/www.haskell.org/builder/default.nix:27:21
(use '--show-trace' to show detailed location information)
~/d/f/www.haskell.org ❯❯❯ nix --version
nix (Nix) 2.3.14```

**Comment by tomjaguarpaw** [comment]
Thanks for the report! @TikhonJelvis or @rebeccaskinner: any ideas?

**Comment by rebeccaskinner** [comment]
I'm not sure off hand, it might be due to macos defaulting to zfs or something? I can try to borrow a mac to test this on soon and try to get a fix up, but if @TikhonJelvis has a mac and can look into it that might be more expedient.

**Comment by TikhonJelvis** [comment]
I only have a Macbook for work, so I haven't tried building the site on a MacOS. I could give it a try, but won't have much time for the next few days at least.

(Honestly, pretty sure I'm going to end up getting an M1 Macbook Air soon for stuff like this + on-the-go development :P.)

**Comment by aviaviavi** [comment]
I just ran into this again today, so just wanted to confirm it's still indeed an open issue and not stale. I still don't know really nix two years later 😭

**Comment by TikhonJelvis** [comment]
Yeah, Nix can be a bit hard to pick up. I believe that we now support a way to build the site entirely without Nix; [have you tried that][1]?

[1]: https://github.com/haskell-infra/www.haskell.org/#manually-building-the-site-with-cabal

**Comment by aviaviavi** [comment]
Yeah I got the project building just fine with cabal, it just takes a while when building from scratch. Not a big deal but definitely a nice-to-have.
